packages/components/src/internal/components/entities/SingleParentEntityPanel.tsx

```diff
@@ -41,14 +41,24 @@ interface Props {
     onChangeParentValue?: (name: string, value: string | any[], index: number) => void;
     onInitialParentValue?: (value: string, selectedValues: List<any>, index: number) => void;
     onRemoveParentType?: (index: number) => void;
-    parentDataType: EntityDataType;
+    parentDataTypes: EntityDataType[];
     parentLSIDs?: string[];
     parentTypeOptions?: List<IEntityTypeOption>;
     parentTypeQueryName?: string;
 }
 
 type SingleParentEntityProps = Props & InjectedQueryModels & OwnProps;
 
+const getChosenTypeSchema = (chosenType: string, parentTypeOptions: List<IEntityTypeOption>): string => {
+
+    const parentDataType = parentTypeOptions
+        .filter(type => {
+            return type.label === chosenType;
```

Comment:
Shouldn't this be using something better to identify the chosen entity? Why not `rowId`?

Reply:
On third look I factored this out, using EntityTypeOption for chosenType which has this info.